### ISSUE TITLE
Failing tests on M1-Arm64 on develop

### ISSUE THREAD

**Issue by NlightNFotis** [diffblue/cbmc#5828]
CBMC version: 5.23.0 (cbmc-5.21.3-exp-254-g268f8864e)
Operating system: macOS 11.0
Exact command line resulting in the issue: `cbmc/regression/ansi-c/gcc_version1 $ clang gcc-5.c`
What behaviour did you expect: Binary is compiled without issue.
What happened instead: Clang fails with error: `__float128 is not supported on this target`.

---

Running the regression test-suite under make on macOS Big Sur, on top of the new m1 platform is causing some of the regression tests to fail. Specifically, `regression/ansi-c/gcc-version1` and `regression/ansi-c/gcc-version2` are failing.

On closer inspection, invoking clang directly on the binary gives me this:

```sh
fotiskoutoulakis@Fotiss-Mac-mini gcc_version1 % clang gcc-5.c
gcc-5.c:10:1: error: __float128 is not supported on this target
__float128 f128;
^
```

Querying clang about version and target produces this:

```sh
fotiskoutoulakis@Fotiss-Mac-mini gcc_version1 % clang -v 
Apple clang version 12.0.0 (clang-1200.0.32.27)
Target: arm64-apple-darwin20.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

I have two questions/remarks:

* It seems like it is assumed that `__float128` is going to be supported on other platforms, but this looks like an unsafe assumption. Should we change the test/deactivate it on some platforms?
* It looks like a subset of the tests in `regression/ansi-c` are marked `gcc-only`. Do we need to run them under MacOS where the default compiler is going to be `clang/llvm`? If not we may need to make changes to our CI infrastructure, as we currently *do* run them under the `macOS-cmake` and `macOS-make` builds.

**Comment by NlightNFotis** [comment]
@tautschnig I'd appreciate your input on this one.

**Comment by tautschnig** [comment]
We had a first round of fixes in 4435150d89, but it seems more work is due. Will take a look.

**Comment by tautschnig** [comment]
@NlightNFotis Could you please try out the fix proposed in #5829? I'd don't have access to an M1 myself, and GitHub doesn't yet support them either (https://github.com/actions/virtual-environments/issues/2187).

**Comment by NlightNFotis** [comment]
@tautschnig I run the test suite again and everything is passing 👍🏻
